main/src/library/Concurrent/Channel.flix

```diff
@@ -166,14 +170,12 @@ namespace Concurrent/Channel {
 
         // If the channel is unbuffered, wait for the element to be handed off
         // before continuing.
-        /* TODO: Optimization possibility: This await could be faster with a
-                 separate condition such that the hand-off signal does not have
-                 to fight for the channel lock with the threads that actually
-                 want to insert an element (unblocking from await requires that
-                 you re-aquire the lock).
-        */
-        if (unBuffered) awaitCondition(waitingPutters)
-        else ();
+        if (unBuffered) {
```

Comment:
It might, at some point make sense to split the data into buffered/unbuffered

Reply:
Yes, agreed. But I wanted to keep the changes for this as simple as possible :-)